csrc/transform_replay.cpp

```diff
@@ -554,9 +554,10 @@ std::pair<TensorDomain*, size_t> TransformReplay::replayPasC(
           consumer->toString(),
           " to producer tensor ",
           producer->toString());
-      new_allocation_domain.emplace_back(it->second);
+      new_allocation_domain.push_back(it->second);
     }
-    replayed->setAllocationDomain(std::move(new_allocation_domain), true);
+    replayed->setAllocationDomain(
+        std::move(new_allocation_domain), consumer->getContiguity());
```

Comment:
I made this change as @jjsjann123 recommended. But I didn't manage to find a good way to test this. Unlike CasP which is used by `TensorView::cache*`, PasC is rarely used by the public API.

Reply:
More specifically, if `producer` and `consumer` have different ranks (e.g., when they are the input and the output of a reduction), `replayed` will get a wrong allocation domain. Can this happen at all? If so, I don't even know what allocation domain to assign -- where do we place the size-1 reduction IterDomain? (I guess the place of the size-1 dimension doesn't matter much in practice, but we'll still have to make a reasonable decision...)

Reply:
I'll look more into this tomorrow. I deliberately added a `NVF_CHECK` asserting producer's rfactor rank == consumer's root rank and noticed several tests including `DynamicTransformIssue418_CUDA` are failing. I can probably revise one of them to get a regression test...

Reply:
Should we follow the same protocol as I commented in https://github.com/NVIDIA/Fuser/pull/1291#discussion_r1393149562 for this?

Reply:
Xiang's comment makes sense to me.

replay on either direction should handle the same issue consistently.

Reply:
This is apparently dead code... (I'm running our unit tests to verify) It appears that never call `TransformReplay::replayPasC` with `replayAllocation` on. Therefore, the function early exits at line 528.

Reply:
All right. I removed the dead code so it's not a concern **at this moment**.